### PR TITLE
[DI] Fix possible incorrect php-code when dumped strings contains newlines

### PR DESCRIPTION
**Pull request by Strate**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | ?
| License       | MIT
| Doc PR        | no

See discussion #24517

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
I think you can still create a simple test case that uses a "\n" and ensure the dumped result is OK (without trying to actually fit that with the case where it is required).

**Comment by Strate** [comment]
@nicolas-grekas done

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -1574,6 +1574,13 @@ private function export($value)
             return $dirname;
         }
 
+        if (is_string($value) && false !== strpos($value, "\n")) {
+            $cleanParts = explode("\n", $value);
+            $cleanParts = array_map(function ($part) { return var_export($part, true); }, $cleanParts);
+
+            return implode(' . "\n" . ', $cleanParts);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: spaces around dots should be remove I think, per our CS policy.

Comment (Strate) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: @nicolas-grekas sure, already done.

**Comment by Strate** [comment]
Some optimizations could be done, for example now string like `"\n\n\n"` dumped as `''."\n".''."\n".''."\n".''`. Is it reasonable to improve that case?

**Comment by nicolas-grekas** [comment]
PHP 7 will concat at compile time, so no need IMHO.

**Review by nicolas-grekas — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Comment by stof** [comment]
So `var_export` does not play well with them ? what about `\r` then ?

**Comment by nicolas-grekas** [comment]
`\r` is fine because the real issue is "explode" in "wrapServiceConditionals", which re-indents the code at "\n" points only.

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @Strate.

**Pull request closed without merging**